web/react/components/admin_console/user_item.jsx

```diff
@@ -76,28 +113,52 @@ export default class UserItem extends React.Component {
         );
     }
 
-    handleMakeSystemAdmin(e) {
+    handleResetPassword(e) {
         e.preventDefault();
+        this.props.doPasswordReset(this.props.user);
+    }
+
+    handleDemote(user, role) {
+        this.setState({
+            serverError: this.state.serverError,
+            showDemoteModal: true,
+            user,
+            role
+        });
+    }
+
+    handleDemoteCancel() {
+        this.setState({
+            serverError: null,
+            showDemoteModal: false,
+            user: null,
+            role: null
+        });
+    }
+
+    handleDemoteSubmit() {
         const data = {
             user_id: this.props.user.id,
-            new_roles: 'system_admin'
+            new_roles: this.props.role
```

Comment:
This prop doesn't seem to make sense and it looks like it will always be an empty string or null since the parent component doesn't pass it in. I think it was meant to be `this.state.role`? Also don't forget to remove the prop declaration at the bottom of the file if I'm correct